packages/dds/tree/src/test/simple-tree/api/snapshotCompatibilityChecker.spec.ts

```diff
@@ -99,4 +102,50 @@ describe("snapshotCompatibilityChecker", () => {
 		// this means these two versions of the application cannot collaborate on content using these schema.
 		assert.equal(forwardsCompatibilityStatus.canView, false);
 	});
+
+	it("checkCompatibility: allowUnknownOptionalFields", () => {
+		const factory = new SchemaFactoryBeta("test");
+
+		// Point2D is constructed with allowUnknownOptionalFields, so it can read Point3D trees
+		// even though it does not know about the optional field `z`.
+		class Point2D extends factory.object(
+			"Point",
+			{
+				x: factory.number,
+				y: factory.number,
+			},
+			{ allowUnknownOptionalFields: true },
+		) {}
+		class Point3D extends factory.object("Point", {
+			x: factory.number,
+			y: factory.number,
+			z: factory.optional(factory.number),
+		}) {}
+
+		const oldViewSchema = new TreeViewConfiguration({ schema: Point2D });
+		const currentViewSchema = new TreeViewConfiguration({ schema: Point3D });
+
+		// Test what the old version of the application would do with a tree using the new schema:
+		const forwardsCompatibilityStatus = checkCompatibility(currentViewSchema, oldViewSchema);
+
+		// Point2D is forwards compatible with Point3D trees
+		assert.equal(forwardsCompatibilityStatus.canView, true);
+	});
+
+	it("checkCompatibility: staged schema", () => {
+		const factory = new SchemaFactoryBeta("test");
+		const oldSchema = factory.optional(
+			factory.types([numberSchema, factory.staged(stringSchema)]),
+		);
+		const currentSchema = factory.optional(factory.types([stringSchema, numberSchema]));
+
+		const oldViewSchema = new TreeViewConfiguration({ schema: oldSchema });
+		const currentViewSchema = new TreeViewConfiguration({ schema: currentSchema });
+
+		// Check to see if the document created by the historical view schema can be opened with the current view schema
+		const backwardsCompatibilityStatus = checkCompatibility(oldViewSchema, currentViewSchema);
```

Comment:
As I noted in https://github.com/microsoft/FluidFramework/pull/25883/files/c20ae4a3906ae61ed52d7194c988d61fedb83661#r2535327047 :

> Testing both compat directions in both these tests would probably be best as thats what users of these features care about: there is no point in trying to allow your old version to collab with the new version if the new version can't collab with the old version: we need both directions for it to offer value.